File: olympic-hcm/2025-olympic10/nangluong/nangluong.py
import os
import sys

input_file = os.path.join(sys.path[0], 'nangluong.inp')
output_file = os.path.join(sys.path[0], 'nangluong.out')

n = a = b = 0
p = 0 # số ngày luyện tập cần tìm


def input_data():
    global n, a, b

    with open(input_file, 'r') as f:
        n = int(f.readline())
        a = int(f.readline())
        b = int(f.readline())


def process():
    global n, b, b, p

    numerator = n * a
    denominator = a + b

    if numerator % denominator == 0:
        p = int(numerator / denominator)
    else:
        p = -1


def output():
    global p

    with open(output_file, 'w') as f:
        f.write(str(p))


if __name__ == '__main__':
    input_data()
    process()
    output()

File: olympic-hcm/2025-olympic10/nangluong/nangluong.cpp
#include <iostream>

#define input_file "nangluong.inp"
#define output_file "nangluong.out"

using namespace std;

int n, a, b;
int p; // số ngày luyện tập cần tìm


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie();

    freopen(input_file, "r", stdin);

    cin >> n >> a >> b;
}


void process()
{
    int numerator = n * a; 
    int denominator = a + b;

    if (numerator % denominator == 0)
    {
        p = numerator / denominator;
    }
    else
    {
        p = -1;
    }
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << p;
}


int main()
{
    input();
    process();
    output();

    return 0;
}